### ISSUE TITLE
Build CUDA with AVX

### ISSUE THREAD

**Issue by martindevans** [SciSharp/LLamaSharp#605]
Currently the CUDA libraries are built with one particular AVX level (it's not specified in the build script, so whatever the default is). This means that models split between GPU and CPU run slower than on the CPU that they could do.

Rather than building:
 - CUDA11
 - CUDA12

We could build
 - noavx/CUDA11
 - noavx/CUDA12
 - AVX/CUDA11
 - AVX/CUDA12
 - AVX2/CUDA11
 - AVX2/CUDA12
 - AVX512/CUDA11
 - AVX512/CUDA12

With the obvious disadvantage that this is a huge explosion in the number of distributed files!

Discussion needed if we actually want this.

**Comment by AsakusaRinne** [comment]
I completely agree to distribute cuda backends with avx. Since we are now distributing cpu and cuda backend, shall we exchange the order of them, like below?

- CUDA11/noavx
- CUDA11/avx
- CUDA11/avx2
- CUDA11/avx512
- CUDA12/noavx
- CUDA12/avx
- CUDA12/avx2
- CUDA12/avx512

**Comment by zsogitbe** [comment]
This is at least 25MBs per cuda setup... 8x25MB = 200 MB just for this above...
I really do not think that this is the right way to go...

We could configure this automatically and let them compile the version they want: I have prepared for this the most important part already which allows to generate a VS solution for both C# and C++: https://github.com/SciSharp/LLamaSharp/pull/628

**Comment by martindevans** [comment]
No longer needed, since llama.cpp has split things up more on it's side.
